### PR TITLE
[BD-46] fix: Segment event catches empty strings

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

- [x] Ensure the openedx.paragon.icons-table.search Segment event contains the search query that was just executed.
- [x] Adding the number of search results given the specified query as an additional event property for this Segment event.

Issue: https://github.com/openedx/paragon/issues/1936

### Deploy Preview

[Icons page](https://deploy-preview-2011--paragon-openedx.netlify.app/foundations/icons)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: www/src/components/IconsTable.jsx [hunk 1]
```diff
@@ -122,18 +109,20 @@ function IconsTable() {
   }, []);
 
   useEffect(() => {
-    debouncedSetIconsList(
-      ICON_NAMES.filter(name => name.toLowerCase().includes(searchValue.toLowerCase())),
-    );
-  }, [searchValue, debouncedSetIconsList]);
+    const list = ICON_NAMES.filter(name => name.toLowerCase().includes(searchValue.toLowerCase()));
+    if (typeof searchValue === 'string' && searchValue.trim() !== '') {
```
Comment (viktorrusakov) [inline www/src/components/IconsTable.jsx:113]: do we really need to check for `typeof searchValue === 'string'` here? I think it's always a string (even if you type a number into the search field)

Comment (PKulkoRaccoonGang) [reply www/src/components/IconsTable.jsx]: Yes, thanks

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2011&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-02-13+01:42&entry.1917517419=2023-04-21+12:15&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**